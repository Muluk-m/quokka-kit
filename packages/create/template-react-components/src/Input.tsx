import React from 'react'

interface InputProps {
  onChange?: (value: string) => void
  value?: string
}

/**
 * Input
 */
export const Input: React.FC<InputProps> = ({ onChange, value }) => {
  return <input value={value} onChange={e => onChange?.(e.target.value)} />
}
