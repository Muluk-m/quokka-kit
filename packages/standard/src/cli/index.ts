#!/usr/bin/env node
import process from 'node:process'
import { spawn } from 'node:child_process'
import c from 'picocolors'
import { hideBin } from 'yargs/helpers'
import yargs from 'yargs'
import * as p from '@clack/prompts'
import { run } from './run'
import { pkgJson } from './constants'

function header() {
  // eslint-disable-next-line no-console
  console.log('\n')
  p.intro(`${c.green(`quokka-standard `)}${c.dim(`v${pkgJson.version}`)}`)
}

const instance = yargs(hideBin(process.argv))
  .scriptName('quokka-standard')
  .usage('')
  .command(
    'commitlint',
    'Run the commit message lint',
    args => args,
    async () => {
      try {
        const child = spawn('pnpm', ['commitlint', '--edit', '$1'], { stdio: 'inherit', shell: true })

        child.on('error', () => {
          process.exit(1) // 直接退出进程
        })

        child.on('exit', (code) => {
          if (code !== 0)
            process.exit(code as number)
        })
      }
      catch {
        process.exitCode = 1
      }
    },
  )
  .command(
    '*',
    'Run the initialization or migration',
    args => args
      .option('yes', {
        alias: 'y',
        description: 'Skip prompts and use default values',
        type: 'boolean',
      })
      .option('template', {
        alias: 't',
        description: 'Use the framework template for optimal customization: vue / react',
        type: 'string',
      })
      .option('extra', {
        alias: 'e',
        array: true,
        description: 'Use the extra utils: formatter / perfectionist / unocss',
        type: 'string',
      })
      .help(),
    async (args) => {
      header()
      try {
        await run(args)
      }
      catch (error) {
        p.log.error(c.inverse(c.red(' Failed to migrate ')))
        p.log.error(c.red(`✘ ${String(error)}`))
        process.exit(1)
      }
    },
  )
  .showHelpOnFail(false)
  .alias('h', 'help')
  .version('version', pkgJson.version)
  .alias('v', 'version')

void instance
  .help()
  .argv
