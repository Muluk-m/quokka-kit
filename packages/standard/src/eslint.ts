import { antfu } from '@antfu/eslint-config'
import defu from 'defu'
import type {
  Awaitable,
  OptionsConfig,
  TypedFlatConfigItem,
} from '@antfu/eslint-config'
import type { Linter } from 'eslint'
import type { FlatConfigComposer } from 'eslint-flat-config-utils'

/**
 * Construct an array of ESLint flat config items.
 *
 * @param {OptionsConfig & TypedFlatConfigItem} options
 *  The options for generating the ESLint configurations.
 * @param {Awaitable<TypedFlatConfigItem | TypedFlatConfigItem[]>[]} userConfigs
 *  The user configurations to be merged with the generated configurations.
 * @returns {Promise<TypedFlatConfigItem[]>}
 *  The merged ESLint configurations.
 */
export function eslint(
  options: OptionsConfig & TypedFlatConfigItem = {},
  ...userConfigs: Awaitable<TypedFlatConfigItem | TypedFlatConfigItem[] | FlatConfigComposer<any>>[]
): FlatConfigComposer<TypedFlatConfigItem> {
  const merged = defu(
    options,
    {
      // Custom Base Config
      astro: false,
      solid: false,
      svelte: false,
    } satisfies OptionsConfig,
  )

  return antfu(merged, {
    rules: {
      'no-console': 'warn',
      'node/prefer-global/buffer': 'off',
      'perfectionist/sort-objects': 'off',
      'react/no-children-prop': 'off',
      'ts/ban-types': 'off',
      'unicorn/prefer-number-properties': 'off',
      'unicorn/prefer-node-protocol': 'off',
      'antfu/top-level-function': 'off',
    },
  }, ...userConfigs)
}

export * from '@antfu/eslint-config'
