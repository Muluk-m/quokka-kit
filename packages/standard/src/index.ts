import { eslint } from './eslint'

export * from './eslint'

export default eslint
