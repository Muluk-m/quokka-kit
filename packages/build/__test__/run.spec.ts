import { resolve } from 'node:path'
import { expect, it } from 'vitest'
import { importConfig } from '../src'

const fixtureDir = resolve(__dirname, 'fixtures')

it('config', async () => {
  const cwd = resolve(fixtureDir, 'config')
  const result = await importConfig(cwd, {
    dts: false,
    vue: true,
    alias: {
      '@': './src',
    },
  })

  expect(result)
    .toMatchSnapshot()
})
