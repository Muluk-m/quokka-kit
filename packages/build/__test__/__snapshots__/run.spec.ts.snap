// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`config 1`] = `
[
  {
    "alias": {
      "@": "./src",
    },
    "dts": false,
    "entry": [
      "src/index.ts",
    ],
    "format": [
      "cjs",
    ],
    "vue": true,
  },
  {
    "alias": {
      "@": "./src",
    },
    "dts": false,
    "entry": [
      "src/config.ts",
    ],
    "vue": true,
  },
]
`;
